pyanaconda/modules/boss/install_manager/install_manager.py

```diff
@@ -39,6 +40,28 @@ def on_module_observers_changed(self, observers):
         """Set module observers which will be used for installation."""
         self._module_observers = list(observers)
 
+    def collect_requirements(self):
+        """Collect requirements of the modules.
+
+        :return: a list of requirements
+        """
+        requirements = []
+
+        for observer in self._module_observers:
+            if not observer.is_service_available:
+                log.warning("Module %s not available!", observer.service_name)
```

Comment:
I wonder - when can this happens & isn't it fatal ? Or do we crash later on elsewhere ?

Reply:
The boss always skips unavailable modules at this moment. In theory, you might call `Quit` for some of them to disable them in the runtime. We can always change this behaviour if it starts to cause problems.

Reply:
Interesting, thanks for the explanation. :)